### DISCUSSION TITLE
Pulling data in main loop

### DISCUSSION THREAD

**Discussion by Ssabaker** [tobozo/ESP32-USB-Soft-Host#45] [General]
Been struggling through trying to get data from a mouse for a while now and came across this library and was excited to finally see data streaming through!  The issue that I'm having now is figuring out how to use the data that the program has parsed. 

For my project I'm trying to gather the x,y data from a mouse and send it over bluetooth (I saw that there may be a separate issue with BT but I'll cross that bridge later). In the sample code I'm struggling to figure out how to actually access the data that I see being displayed on the serial monitor. Is there a specific function call for this or is the process locked in a loop from the way that the USH.init is called? I tried a test Serial.print in the main loop but it seems that it never runs that portion of the program.

This might be something obvious that I'm overlooking here but any help would be appreciated, thanks.

**Comment by tobozo** [comment]
you need to typecast `uint8_t` to `int8_t` to get significant x/y/wheel values from the data report, those are generally signed vectors 

USH.init [third argument](https://github.com/tobozo/ESP32-USB-Soft-Host/blob/main/examples/USB_Test/USB_Test.ino#L124) is a function callback that gets fired every time the HID device sends report data, this is where you should enqueue the packet you will later forward to BT. This function is called from an interrupt so you should avoid doing blocking stuff from there.
